src/map/pc.cpp

```diff
@@ -5082,6 +5083,93 @@ bool pc_skill(struct map_session_data* sd, uint16 skill_id, int level, enum e_ad
 	}
 	return true;
 }
+
+/**
+ * Set's a player's plagiarized skill.
+ * @param sd: Player
+ * @param skill_id: Skill to be plagiarized
+ * @param skill_lv: Skill level to be plagiarized
+ * @return True on success or false otherwise
+ */
+bool pc_skill_plagiarism(map_session_data &sd, uint16 skill_id, uint16 skill_lv)
+{
+	uint16 idx;
+	int type = skill_isCopyable(&sd, skill_id);
+
+	skill_id = skill_dummy2skill_id(skill_id);
+	skill_lv = cap_value(skill_lv, 1, skill_get_max(skill_id));
+
+	//Use skill index, avoiding out-of-bound array [Cydh]
+	if (!(idx = skill_get_index(skill_id))) {
+		ShowWarning("pc_skill_plagiarism: invalid skill idx %d for skill %d.\n", idx, skill_id);
```

Comment:
Well idx is always 0 in this case, the line could be straight (skill_get_index need to be above)
```suggestion
	if (idx == 0) {
		ShowWarning("pc_skill_plagiarism: invalid skill idx 0 for skill %d.\n", skill_id);
```